backend/app/controllers/stats_controller.rb

```diff
@@ -2,6 +2,17 @@
 
 class StatsController < ApplicationController
   def index
+    redemptions = Redemption.includes(:coupon, campaign_charity: :charity).all
+
+    contribution_amount_from_primary_donor = redemptions.map { |r| r.coupon.denomination }.sum
+    contribution_amount_from_secondary_donors = SecondaryDonation.sum(:amount)
+    @total_contribution_amount = contribution_amount_from_primary_donor + contribution_amount_from_secondary_donors
+
+    @total_redemption_count = redemptions.count
+    @total_charities_supported = redemptions.map { |r| r.campaign_charity.charity_id }.uniq.count
```

Comment:
```suggestion
    @total_charities_supported = CampaignCharity.distinct.count(:charity_id)
```

The number of charities we support should not depend on the redemptions? Then we can remove the `campaign_charity` preload.